rosidl_generator_cpp/test/test_array_generator.hpp

```diff
@@ -139,4 +140,42 @@ void test_vector_fill(C * container, size_t size,
   }
 }
 
+/**
+ * Helper function to generate a test pattern for string types.
+ * Mininum and maximum values for the type and random numbers in the middle.
+ * @param C Container (vector, array, etc) to be filled
+ * @param size How many elements to fill in. Must size<=container_size
+ * @param min Minimum value in the range to fill.
+ * @param max Maximum value in the range to fill.
+ * @param minlength Minimum length of the generated strings.
+ * @param maxlength Maximum length of the generated strings.
+ */
+template<
+  typename C,
+  typename std::enable_if<
+    std::is_same<typename C::value_type, typename std::string>::value
+  >::type * = nullptr
+>
+void test_vector_fill(C * container, size_t size,
+  int min, int max,
+  int minlength, const int maxlength)
+{
+  std::default_random_engine rand_generator;
+  std::uniform_int_distribution<int> randnum(min, max);
+  std::uniform_int_distribution<int> randlen(minlength, maxlength);
```

Comment:
I would recommend not to use random numbers for any tests. The problem is that the test will be flaky of the result depends on the values. Instead use a "known" pattern, e.g. increment numbers, or ascii characters.